### PR TITLE
cmd/geth: allow for big ttd

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**

### CODE REVIEW

File: cmd/geth/config.go [hunk 1]
```diff
@@ -160,7 +160,7 @@ func makeFullNode(ctx *cli.Context) (*node.Node, ethapi.Backend) {
 		cfg.Eth.OverrideArrowGlacier = new(big.Int).SetUint64(ctx.GlobalUint64(utils.OverrideArrowGlacierFlag.Name))
 	}
 	if ctx.GlobalIsSet(utils.OverrideTerminalTotalDifficulty.Name) {
-		cfg.Eth.OverrideTerminalTotalDifficulty = new(big.Int).SetUint64(ctx.GlobalUint64(utils.OverrideTerminalTotalDifficulty.Name))
+		cfg.Eth.OverrideTerminalTotalDifficulty, _ = new(big.Int).SetString(ctx.GlobalString(utils.OverrideTerminalTotalDifficulty.Name), 0)
```
Comment (rjl493456442) [inline cmd/geth/config.go:163]: Might be worthwhile to check the validity of given ttd. If it's invalid the returned big.Int can be nil.

**Review by rjl493456442 — APPROVED** [review]
nitpick, LGTM

**Review by Mitumaru — APPROVED** [review]

**Comment by MariusVanDerWijden** [comment]
I added another commit to fix a panic that happens with BigFlags, because we try to set the value of a non-initialized field
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe6f9ca]

goroutine 1 [running]:
github.com/ethereum/go-ethereum/cmd/utils.(*bigValue).Set(0x0, {0x7ffecd425998, 0x0})
	github.com/ethereum/go-ethereum/cmd/utils/customflags.go:161 +0x4a
flag.(*FlagSet).parseOne(0xc0000bef60)
	flag/flag.go:986 +0x650
flag.(*FlagSet).Parse(0xc0000bef60, {0xc0001ac010, 0xc00019e000, 0x9e})
	flag/flag.go:1005 +0x4f
gopkg.in/urfave/cli%2ev1.(*App).Run(0xc000534d00, {0xc0001ac000, 0xe, 0xe})
	gopkg.in/urfave/cli.v1@v1.20.0/app.go:193 +0x185
main.main()
	github.com/ethereum/go-ethereum/cmd/geth/main.go:269 +0x35
```

File: cmd/utils/customflags.go [hunk 2]
```diff
@@ -154,11 +154,11 @@ func (b *bigValue) String() string {
 }
 
 func (b *bigValue) Set(s string) error {
-	int, ok := math.ParseBig256(s)
-	if !ok {
+	intVal, ok := math.ParseBig256(s)
+	if !ok || intVal == nil {
```
Comment (NamanJain8) [inline cmd/utils/customflags.go:158]: `intVal == nil` check is not needed, right? Ideally, anything that got parsed successfully should return a non-nil value.

Comment (fjl) [reply cmd/utils/customflags.go]: fixed

**Pull request merged**